plugins/routed-eni/cni.go

```diff
@@ -182,6 +180,15 @@ func add(args *skel.CmdArgs, cniTypes typeswrapper.CNITYPES, grpcClient grpcwrap
 	return cniTypes.PrintResult(result, cniVersion)
 }
 
+// generateHostVethName returns a name to be used on the host-side veth device.
+// The veth name is generated such that it aligns with the value expected
+// by Calico for NetworkPolicy enforcement.
+func generateHostVethName(namespace, podname string) string {
+	h := sha1.New()
+	h.Write([]byte(fmt.Sprintf("%s.%s", namespace, podname)))
+	return fmt.Sprintf("cali%s", hex.EncodeToString(h.Sum(nil))[:11])
```

Comment:
I think "eni" is better prefix than "cali" because  this is a virtual interface over  AWS VPC ENI (Elastic Network interface).

Reply:
From the comment, it looks like you depend on the "cali" prefix to discover interfaces for Network Policy Enforcement.  Is there another mechanism calico could use (configuration or API) to do this discovery that's not dependent on a convention that requires us to name this in a misrepresentative way.

Reply:
Yes - Ideally the prefix defaults to eni and can be overridden by a config flag/env variable. We can have a seperate cni daemon set file that carries the "cali" prefix as a flag/env.

Reply:
I think using `cali` as the prefix provides a number of benefits, namely:

- This is the standard naming convention used across all other Calico deployments, and so it's what users expect.
- Similarly, existing documentation, guides and troubleshooting documentation reference the `cali` prefix, so choosing this prefix will allow all of that material to "just work".

There's another benefit of doing this by default, which is that it allows users to enable Calico for NetworkPolicy at any point in time, not just at cluster creation. This is really valuable for users who might create a cluster without the feature enabled, and then realize they need it later on.

Reply:
This seems to be the inverse of what I'd expect - I don't think of Calico as the defining standard for networking in Kubernetes, but rather one of many options. If Calico is setting out to be an generic add-on to enforce NetworkPolicy on any underlying CNI network, would it not make more sense for it to handle the varying cases that underlying CNI networks might present better? This is something we also faced with implementing NetworkPolicy for our AWS/CNI arrangement. Similar goes for mapping pods to interfaces.

Reply:
It doesn't make sense to me to put cali in AWS/CNI. I may run the plugin in AWS without Calico involved but have a different implementation for network policy.

Reply:
@caseydavenport The first two callouts would still stand even with the recommended change using "cali" as the prefix flag/env. Like others have chimed in, its important that this CNI plugin works in a standalone environment for which it doesn't make sense to have "cali" as prefix. 

I understand this would require customers to decide at launch time. That is solvable with Calico supporting a configurable prefix as well.